src/librustc/ty/subst.rs

```diff
@@ -30,7 +31,7 @@ use std::num::NonZeroUsize;
 /// To reduce memory usage, a `Kind` is a interned pointer,
 /// with the lowest 2 bits being reserved for a tag to
 /// indicate the type (`Ty` or `Region`) it points to.
-#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash)]
```

Comment:
I don't really think `PartialOrd` deriving calls into `Ord::cmp`, but rather it compares the fields.
So when you need a custom `impl` for `Ord`, you also need to implement `PartialOrd`.
But if you have `Ord::cmp`, you can just call it from `PartialOrd::partial_cmp`.